pandas/core/arrays/datetimelike.py

```diff
@@ -1645,6 +1645,24 @@ def _with_freq(self, freq):
         arr._freq = freq
         return arr
 
+    # --------------------------------------------------------------
+
+    def factorize(self, na_sentinel=-1, sort: bool = False):
```

Comment:
_values_for_factorize and _from_factorized in DatetimeLikeArrayMixin are now only used by PeriodArray? should these be moved?

Reply:
values_for_factorize is still used (incorrectly) in core.util.hashing, so for now this is still needed here

Reply:
ok